packages/pwa/app/commerce-api/hooks/useCustomerProductList.js

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useContext, useMemo} from 'react'
+import {useCommerceAPI, CustomerProductListsContext} from '../contexts'
+import {isError} from '../utils'
+
+import useCustomer from './useCustomer'
+
+/**
+ * This hook is designed to add customer product list capabilities
+ * to your app, it leverages the Commerce API - Shopper Customer endpoints.
+ * It uses a React context to store customer product list globally.
+ *
+ * By default, the Shopper Customer Product List API allows a shopper to
+ * save multiple product lists. However, the PWA only use a single
+ * product list, which is the wishlist. There is another hook useCustomerProductList,
+ * which is built on top of this hook to add wishlist specific logic.
+ * You can think this hook as the "parent class" for the useCustomerProductList hook.
+ *
+ * If your application need to handle multiple customer product lists, this
+ * is the hook for you, otherwise it is recommended to use useCustomerProductList hook.
```

Comment:
This might be forward looking (when we refactor into the sdk or other package), but we'll have to mention that this hook will only work within the context of the `CustomerProductListProvider`.